java/java.lsp.server/src/org/netbeans/modules/java/lsp/server/explorer/NodePropertiesProvider.java

```diff
@@ -0,0 +1,210 @@
+
+    private static final String COMMAND_GET_NODE_PROPERTIES = "java.node.properties.get";      // NOI18N
+    private static final String COMMAND_SET_NODE_PROPERTIES = "java.node.properties.set";      // NOI18N
+
+    private static final String PROP_NAME = "propName";      // NOI18N
+    private static final String PROP_DNAME = "propDispName";      // NOI18N
+    private static final String PROP_HTML_NAME = "propHtmlName";      // NOI18N
+    private static final String PROP_SHORT_NAME = "propShortName";      // NOI18N
+    private static final String PROP_PREF = "propPref";      // NOI18N
+    private static final String PROP_EXPERT = "propExpert";      // NOI18N
+    private static final String PROP_HIDDEN = "propHidden";      // NOI18N
+    private static final String PROP_CAN_READ = "propRead";      // NOI18N
+    private static final String PROP_CAN_WRITE = "propWrite";      // NOI18N
+    private static final String PROP_VAL_TYPE = "propType";      // NOI18N
+    private static final String PROP_VALUE = "propValue";      // NOI18N
+    private static final String PROPS = "props";      // NOI18N
+
+    private static final Set<String> COMMANDS = new HashSet<>(Arrays.asList(
+            COMMAND_GET_NODE_PROPERTIES, COMMAND_SET_NODE_PROPERTIES
+    ));
+
+    private final Gson gson = new Gson();
+
+    @Override
+    public List<CodeAction> getCodeActions(ResultIterator resultIterator, CodeActionParams params) throws Exception {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public CompletableFuture<Object> processCommand(NbCodeLanguageClient client, String command, List<Object> arguments) {
+        if (!COMMANDS.contains(command)) {
+            return CompletableFuture.completedFuture(null);
+        }
+        if (arguments == null || arguments.isEmpty()) {
+            return CompletableFuture.completedFuture(null);
+        }
+        TreeNodeRegistry r = Lookup.getDefault().lookup(TreeNodeRegistry.class);
+        if (r == null) {
+            return CompletableFuture.completedFuture(null);
+        }
+        int nodeId = ((JsonPrimitive) arguments.get(0)).getAsInt();
+        TreeViewProvider nodeProvider = r.providerOf(nodeId);
+        Node node = null;
+        if (nodeProvider != null) {
+            node = nodeProvider.findNode(nodeId);
+        }
+        if (node == null) {
+            return CompletableFuture.completedFuture(null);
+        }
+        boolean getProps = COMMAND_GET_NODE_PROPERTIES.equals(command);
+        Node.PropertySet[] propertySets = node.getPropertySets();
+
+        if (getProps) {
+            return CompletableFuture.completedFuture(getAllPropertiesMap(propertySets));
+        }
+        if (arguments.size() == 2) {
+            Object propJson = arguments.get(1);
+            if (propJson instanceof JsonNull) {
+                return CompletableFuture.completedFuture(null);
+            }
+            List m = gson.fromJson((JsonElement) propJson, List.class);
```

Comment:
Hm ... could be more friendly if the property sets was a map (keyed by set name) instead of array, but the propset name is present in each array's item, so it's probably OK.

Maybe if
```
List<Map<String, Object>> m = ...
```
the `setAllProperties` and others can use strongly typed Map instead of raw type(s) ?